### ISSUE TITLE
Remove the qt6 import hack for windows

### ISSUE THREAD

**Issue by data-sync-user** [mozilla-mobile/mozilla-vpn-client#2975]
Qmake qt6 is unable to process the QML resources from QRC files not in the
"root" of the project (`src/src.pro`) and because of this, it's unable to
generate the full list of QT modules/plugins to include in the final static
build.

We do have QRC files in other folders: `glean`, `lottie`, `nebula` are 3
examples of folders with QRC files "outside" the `src` path.

The workaround is to have a separate `qt6winhack.qml` file (and its
`qt6winhack.qrc`) that includes all the required QT modules (`localstorage`,
`qtcharts`, etc).

If you don't care about "why" Qt is buggy, stop reading here.

To be more precise about why this bug happens and why we cannot fix this issue
in a different way, I want to describe here the source of it.

`qmake` uses a Qt tool to parse the QML files and extract the list of modules
to be included in the final build. This tool is called `qmlimportscanner`
(https://github.com/qt/qtdeclarative/tree/dev/tools/qmlimportscanner).

This tool is executed in this way:

```
qmlimportscanner <the/path/of/the/project> ~~importPath </qt/internal/qml/path>~~ qrcFiles <list/of/qrc/resources>
```

on windows, the qrcFiles parameter is a list of
`C:/the/full/path/of/a/file.qrc`. Note 2 things: `C:` and the '/' (instead of
'\').

This tool in 6.2.x is buggy because:

1. qmlimportscanner calls QQmlJSResourceFileMapper to generate the list of
QML/JS files to parse:
https://github.com/qt/qtdeclarative/blob/6.2.4/tools/qmlimportscanner/main.cpp#L704-L707

2. But when QQmlJSResourceFilterMapper filters the QML/JS files, it runs this:
https://github.com/qt/qtdeclarative/blob/6.2.4/src/qmlcompiler/qqmljsresourcefilemapper.cpp#L111-L112 
where `terminatedDirectory` is set to '/' - no filter.path is set!

3. Then, a few lines later, `terminatedDirectory` is used here:
https://github.com/qt/qtdeclarative/blob/6.2.4/src/qmlcompiler/qqmljsresourcefilemapper.cpp#L121-L122
to compare how the `candidate` filename starts.

4. But the `candidate` file is `C:/the/full/path/of/a/file/qml`, it does not
start with '/'.

So, basically, no QML/JS files from QRC files are scanned.

The fix is trivial:
https://github.com/qt/qtdeclarative/blob/dev/src/qmlcompiler/qqmljsresourcefilemapper.cpp#L121
(see
https://github.com/qt/qtdeclarative/commit/7517c1b3ae9aa92f36b19d74a4b2de5e8531309b)
but it's not included in 6.2.x.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/VPN-1894)
┆author: Andrea Marchesini

**Comment by data-sync-user** [comment]
➤ Andrea Marchesini commented:

We still include qt6winhack.qml.I suspect this bug is still valid.
